### PR TITLE
Make non required props optional in typescript definitions

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

When using `TabularMainPage` component I noticed the `title` prop is marked as required in its typescript definition but that prop is actually optional since there is another one (`customTitleRow`) which you can use to tailor the header section of that component instead of just using a _string_.

I also found the same situation with `TabularDetailPage` component.

#### Description

This PR updates both components props TS definition to set `title` as optional and have am internal check so either that one or `customTitleRow` are provided when using the components.

### CODE REVIEW

File: .changeset/neat-trees-itch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-components': patch
+---
+
+Update both `TabularMainPage` and `TabularDetailPage` components so its `title` prop is not required in their typescript definitions as they are actually optional in their implementation.
```
Comment (emmenko) [inline .changeset/neat-trees-itch.md:5]:
```suggestion
Make the `title` prop optional for the components `<TabularMainPage>` and `<TabularDetailPage>`.

A warning will be logged if neither the `title` or the `customTitleRow` props are provided.
```

File: packages/application-components/src/components/tabular-detail-page/tabular-detail-page.tsx [hunk 2]
```diff
@@ -29,7 +30,7 @@ type TTabularDetailPageProps = {
   /**
    * The title of the page.
    */
-  title: string;
+  title?: string;
```
Comment (emmenko) [inline packages/application-components/src/components/tabular-detail-page/tabular-detail-page.tsx:33]: Can you also update the docs (prop types table) to make this optional?

We don't have this automated in appkit yet. Thanks

**Review by emmenko — COMMENTED** [review]
Thanks, I'm fine then with this.

File: packages/application-components/src/components/tabular-main-page/tabular-main-page.tsx [hunk 3]
```diff
@@ -51,40 +52,47 @@ const defaultProps: Pick<TTabularMainPageProps, 'hideControls'> = {
...
-  <PageWrapper>
-    <TabularPageContainer color="surface">
-      <Spacings.Stack>
-        {props.customTitleRow || (
-          <PageHeaderTitle
-            title={props.title}
-            subtitle={props.subtitle}
-            titleSize="big"
+const TabularMainPage = (props: TTabularMainPageProps) => {
+  warning(
```
Comment (kark) [inline packages/application-components/src/components/tabular-main-page/tabular-main-page.tsx:56]: nit: we could also add tests for this case - I attempted doing that [at some point](https://github.com/commercetools/merchant-center-application-kit/blob/bf93463ebd4cb8cf80294dcaaedb115e7adeb8f5/packages/application-components/src/components/tabular-detail-page/tabular-detail-page.spec.tsx#L112:L121)

**Review by kark — COMMENTED** [review]
Looking good! 👍

File: packages/application-components/src/components/tabular-main-page/tabular-main-page.tsx (hunk 3 above)
Comment (CarlosCortizasCT) [reply packages/application-components/src/components/tabular-main-page/tabular-main-page.tsx:56]: Sure.
I added a new test to both components.

Thanks for the suggestion 👍

**Review by emmenko — APPROVED** [review]

**Review by kark — APPROVED** [review]

**Pull request merged**